pkg/innerring/internal/blockchain/blockchain.go

```diff
@@ -0,0 +1,383 @@
+		if err != nil {
+			return nil, fmt.Errorf("init BoltDB storage for blockchain: %w", err)
+		}
+	}
+
+	blockChain, err := core.NewBlockchain(blockchainStorage, cfgBase.Blockchain(), opts.Logger)
+	if err != nil {
+		closeErr := blockchainStorage.Close()
+		if closeErr != nil {
+			return nil, fmt.Errorf("init core blockchain component: %w; failed to close blockchain storage: %v", err, closeErr)
+		}
+
+		return nil, fmt.Errorf("init core blockchain component: %w", err)
+	}
+
+	go blockChain.Run()
+
+	netServer, err := network.NewServer(cfgServer, blockChain, blockChain.GetStateSyncModule(), opts.Logger)
+	if err != nil {
+		return nil, fmt.Errorf("init NeoGo network server: %w", err)
+	}
+
+	if !opts.NotaryDisabled {
+		var cfgNotary notary.Config
+		cfgNotary.MainCfg.Enabled = true
+		cfgNotary.MainCfg.UnlockWallet = cfg.Wallet
+		cfgNotary.Chain = blockChain
+		cfgNotary.Log = opts.Logger
+
+		notaryService, err := notary.NewNotary(cfgNotary, netServer.Net, netServer.GetNotaryPool(), func(tx *transaction.Transaction) error {
+			err := netServer.RelayTxn(tx)
+			if err != nil && !errors.Is(err, core.ErrAlreadyExists) {
+				return fmt.Errorf("relay completed notary transaction %s: %w", tx.Hash().StringLE(), err)
+			}
+
+			return nil
+		})
+		if err != nil {
+			return nil, fmt.Errorf("init Notary service: %w", err)
+		}
+
+		netServer.AddService(notaryService)
+		blockChain.SetNotary(notaryService)
+	}
+
+	// "make" channel rw to satisfy Start method
+	chErrRw := make(chan error)
+
+	go func() {
+		err, ok := <-chErrRw
+		if ok {
+			cfg.ErrorListener <- err
+		}
+	}()
+
+	rpcServer := rpcsrv.New(blockChain, cfgBaseApp.RPC, netServer, nil, opts.Logger, chErrRw)
+
+	netServer.AddService(&rpcServer)
+
+	res := &Blockchain{
+		nodeAcc:   nodeAcc,
+		logger:    opts.Logger,
+		core:      blockChain,
+		netServer: netServer,
+		rpcServer: &rpcServer,
+		chErr:     chErrRw,
+		consensusConfig: consensus.Config{
+			Logger:                opts.Logger,
+			Broadcast:             netServer.BroadcastExtensible,
+			Chain:                 blockChain,
+			BlockQueue:            netServer.GetBlockQueue(),
+			ProtocolConfiguration: blockChain.GetConfig().ProtocolConfiguration,
+			RequestTx:             netServer.RequestTx,
+			StopTxFlow:            netServer.StopTxFlow,
+			TimePerBlock:          cfgBaseProto.TimePerBlock,
+			Wallet:                cfg.Wallet,
+		},
+		chStop: make(chan struct{}),
+	}
+
+	return res, nil
+}
+
+// Run runs the Blockchain and makes all its functionality available for use.
+// Returns any error encountered which prevented the Blockchain to be started.
+// If Run failed, the Blockchain should no longer be used. After Blockchain has
+// been successfully run, all internal failures are written to the configured
+// listener.
+//
+// Run should not be called more than once.
+//
+// Use Stop to stop the Blockchain.
+func (x *Blockchain) Run() error {
+	go x.netServer.Start(x.chErr)
+	go x.switchConsensusLoop()
```

Comment:
Maybe we can omit it at all? The consensus service will be in watch-only mode if the node is not elected, so it can always kick in, but it'll do almost nothing if it's not.